src/components/node-list/node-list-tree.js

```diff
@@ -118,6 +127,23 @@ const TreeListProvider = ({
     }
   };
 
+  const onItemExpandToggle = (event, expandedItemIds) => {
+    // Reset clicked node
+    onToggleNodeClicked(null);
+
+    // Contract all modular pipelines that are not expanded on sidebar
+    for (const id of modularPipelineIds) {
+      if (!expandedItemIds.includes(id)) {
+        onToggleContracted(id, true);
+      }
+    }
```

Comment:
As per my comment above this would likely not be needed if all modularPipelines are set as contracted by default in the state?

Reply:
Even with the default state set as you mentioned, I think this is still needed since the material-ui `TreeView` component holds its own internal state. I looked to see if it was possible to drive the `TreeView` state through props but it doesn't seem to expose that.

Reply:
This is also needed because once user collapses a node on the sidebar, we need to dispatch an action to collapse it on the main graph as well.